src/buffer/canvas.rs

```diff
@@ -0,0 +1,639 @@
+use core::ops::Deref;
+use std::io::{Seek, Write};
+
+use crate::{
+    error::{DecodingError, ImageFormatHint},
+    flat::SampleLayout,
+    traits::{EncodableLayout, PixelWithColorType},
+    ColorType, DynamicImage, ExtendedColorType, FlatSamples, ImageBuffer, ImageDecoder, ImageError,
+    ImageOutputFormat, ImageResult, Luma, LumaA, Rgb, Rgba,
+};
+
+use image_canvas::layout::{CanvasLayout as InnerLayout, SampleBits, SampleParts, Texel};
+use image_canvas::Canvas as Inner;
+use image_canvas::{color::Color, layout::Block};
+
+/// A byte buffer for various color and layout combinations.
+///
+/// Note that it is possible to convert most [`ImageBuffer`] as well as [`DynamicImage`] instances
+/// into a canvas that represents the same pixels and color. See `From` implementations below.
+pub struct Canvas {
+    inner: Inner,
+}
```

Comment:
@fintelia Alright with the name? Other ideas that have been floating around in my head:

- `Picture` for a less dumb buffer that also contains the metadata.
- How do we treat the current `ImageBuffer`, I really want to preserve the 'simple' pixel array container because it can be much simpler to work with and requires less unwrapping. For instance, we couldn't offer `rows` on `Canvas`. But the naming is odd now. Maybe `PixelBuffer` is alright?